### DISCUSSION TITLE
Improvement proposal for the example on `pyvista.StructuredGrid`

### DISCUSSION THREAD

**Discussion by natsuwater** [pyvista/pyvista#4360] [Ideas]
Hello, 

Thank you for making such a great library to work on mesh structures.

I found a little confusing behavior on `StructuredGrid`, figured out the reason, and have a proposal to improve the example in the document.

## Code

Here's a minimal code that reproduces the behavior.

```python
import numpy as np
import pyvista as pv
xary = np.arange(-10, 10, 2, dtype=np.float64)
yary = np.arange(-10, 10, 5, dtype=np.float64)
zary = np.arange(-10, 10, 1, dtype=np.float64)

grid_rl = pv.RectilinearGrid(xary, yary, zary)

x, y, z = np.meshgrid(xary, yary, zary)
grid_st = pv.StructuredGrid(x, y, z)

```

The resulting RectilinearGrid `grid_rl` is 10 x 4 x 20 as in:
```
Out[2]:
RectilinearGrid (0x1e7a21991c0)
  N Cells:      513
  N Points:     800
  X Bounds:     -1.000e+01, 8.000e+00
  Y Bounds:     -1.000e+01, 5.000e+00
  Z Bounds:     -1.000e+01, 9.000e+00
  Dimensions:   10, 4, 20
  N Arrays:     0
``` 
Tje StructuredGrid `grid_st` is 4 x 10 x 20:
```
In [3]: grid_st
Out[3]:
StructuredGrid (0x1e7a21b1760)
  N Cells:      513
  N Points:     800
  X Bounds:     -1.000e+01, 8.000e+00
  Y Bounds:     -1.000e+01, 5.000e+00
  Z Bounds:     -1.000e+01, 9.000e+00
  Dimensions:   4, 10, 20
  N Arrays:     0
``` 
The behavior was produced on  `pv.__version__  '0.37.0'`, `np.__version__ '1.24.1'` on windows11.  

This code was combined version of the example on https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.StructuredGrid.html and https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.RectilinearGrid.html

## What's going on

In the StructuredGrid above, the index of the cell along y-axis changes first then index along x-axis incremnets.  So `pyvista` itself is working just right.  But its confusing at least for non-experienced pyvista users.

This actually is due to the implementation of the `numpy.meshgrid` which has an option `indexing ='xy' (default)` and  `indexing ='ij'`.
So if I write `x, y, z = np.meshgrid(xary, yary, zary, indexing="ij")`, the dimension of StructuredGrid becomes 10 x 4 x 20.
Thie is coherent with RectilinearGrid and easier to understand.

## Proposal for documentation update

So I suggest that the example code 
https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.RectilinearGrid.html
would be better with something lile
```python
>>> xrng = np.arange(-10, 10, 2, dtype=np.float32)
>>> yrng = np.arange(-10, 10, 5, dtype=np.float32)
>>> zrng = np.arange(-10, 10, 1, dtype=np.float32)
>>> x, y, z = np.meshgrid(xrng, yrng, zrng, indexing='ij')
>>> grid = pyvista.StructuredGrid(x, y, z)
```

Thank you.

**Comment by tkoyama010** [comment]
@natsuwater Thank you for the idea. I agree with this change. If possible, do you have time to create a PullRequest for us? We will be happy to review and merge.

**Reply by natsuwater** [reply]
@tkoyama010 I'm willing to create a PullRequest, but where is the documentation source for [pyvista.StructuredGrid.html](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.StructuredGrid.html)? 
 I could not find one under https://github.com/pyvista/pyvista/blob/main/doc/source/api/core/

**Reply by tkoyama010** [reply]
@natsuwater Good question, the documentation under the api path is automatically created from the Python source code docstring. This means that to change the documentation, you need to change the docstring below.
https://github.com/pyvista/pyvista/blob/caa1129197fd89f6cb7ad532548d1e56f5405038/pyvista/core/grid.py#L65-L122

**Reply by natsuwater** [reply]
OK. I understand. I'll make a PullRequest.

**Reply by natsuwater** [reply]
I mistyped earlier in this thread. 

My proposal for documentation update was on `StructuredGrid`, not on `RectilinearGrid`.  So, I modified `pointset.py` and submitted a PullRequest.

Should I edit the original post so as to reduce future confusion?

**Reply by natsuwater** [reply]
The PullRequest is  at https://github.com/pyvista/pyvista/pull/4365 . It seems that it should have been reviewed before I pushed the button.
